distributed/submit/remote_cli.py

```diff
@@ -0,0 +1,34 @@
+import socket
+import logging
+
+import click
+
+from tornado.ioloop import IOLoop
+from distributed.submit.remote_client import RemoteClient
+
+from distributed.utils import get_ip
+
+logger = logging.getLogger('distributed.remote')
+
+
+@click.command()
+@click.option('--host', type=str, default=None,
+              help="IP or hostname of this server")
+@click.option('--port', type=int, default=8788, help="Remote Client Port")
+def remote(host, port):
+    _remote(host, port)
+
+
+
+def _remote(host, port, loop=IOLoop.current(), client=RemoteClient):
+    host = host or get_ip()
+    if ':' in host and port == 8788:
+        host, port = host.rsplit(':', 1)
+        port = int(port)
+    ip = socket.gethostbyname(host)
+    remote_client = client(ip=ip, loop=loop)
+    remote_client.start(port=port)
+    loop.start()
+    loop.close()
+    remote_client.stop()
+    logger.info("End remote client at %s:%d", host, port)
```

Comment:
I would have the `_submit` coroutine return values for stderr and stdout and have the CLI code emit those to sys.stderr/sys.stdout.  To me the CLI and sys.stderr/sys.stdout go together while coroutines and returning values go together.  I would keep the test but remove the capsys stuff and just check the return value of `_submit`.  

I have to admit though that I'm not as familiar with what's going on here as you are and I'm not paying a whole lot of attention to this.  I may be giving bad advice.